Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -73,12 +73,21 @@ enum {
 static GParamSpec* sObjProperties[N_PROPERTIES] = { nullptr, };
 
 struct _WebKitDownloadPrivate {
+    ~_WebKitDownloadPrivate()
+    {
+        if (decideDestinationCallback) {
+            g_critical("Bug: application handled WebKitDownload::decide-destination but failed to call webkit_download_set_destination() before the WebKitDownload was destroyed");
+            (*decideDestinationCallback)(AllowOverwrite::No, { });
+        }
+    }
+
     RefPtr<DownloadProxy> download;
 
+    std::optional<CompletionHandler<void(AllowOverwrite, String)>> decideDestinationCallback;
```

Comment:
I don't think we need to use `std::optional` CompletionHandler can be set to nullptr

Reply:
Oops, didn't know.